libsolidity/interface/Natspec.cpp

```diff
@@ -255,3 +271,35 @@ Json::Value Natspec::devDocumentation(std::multimap<std::string, DocTag> const&
 
 	return json;
 }
+
+vector<EventDefinition const*>  Natspec::uniqueInterfaceEvents(ContractDefinition const& _contract)
+{
+	auto eventSignature = [](EventDefinition const* _event) -> string {
+		FunctionType const* functionType = _event->functionType(true);
+		solAssert(functionType, "");
+		return functionType->externalSignature();
+	};
+	auto compareBySignature =
+		[&](EventDefinition const* _lhs, EventDefinition const* _rhs) -> bool {
+			return eventSignature(_lhs) < eventSignature(_rhs);
+		};
+
+	set<EventDefinition const*, decltype(compareBySignature)> uniqueEvents{compareBySignature};
+	// Insert events defined in the contract first so that in case of a conflict
+	// they're the ones that get selected.
+	uniqueEvents += _contract.definedInterfaceEvents();
+
+	set<EventDefinition const*, decltype(compareBySignature)> filteredUsedEvents{compareBySignature};
+	set<string> usedSignatures;
+	for (EventDefinition const* event: _contract.usedInterfaceEvents())
+	{
+		auto&& [eventIt, eventInserted] = filteredUsedEvents.insert(event);
+		auto&& [signatureIt, signatureInserted] = usedSignatures.insert(eventSignature(event));
+		if (!signatureInserted)
+			// Either the old event or the new one will be erased
```

Comment:
Is there a case for two events with the same signature being emitted by a contract that have distinct definitions, neither of which occurs in the emitting contract?

Reply:
``emit_same_signature_event_different_libraries`` in the natspec docs, which indicates that this *intentionally* omits the signature altogether in case of an ambiguity...
That's one method of treating ambiguities, not sure it's the best one - but yeah, that contradicts the comment here in any case :-).

Reply:
I'm fine either way - we can skip the event entirely in this case, but then change the comment here - or choose an arbitrary one in this case, which would mean replacing the code.

Reply:
Haha, I just read through the backlog of discussion on this topic - I'm very glad that I've stayed out of this so far.

Reply:
@matheusaaguiar Just remove the comment here and we'll be done with this sad story :-).